Source/WebCore/dom/Element.cpp

```diff
@@ -2498,6 +2501,47 @@ void Element::didMoveToNewDocument(Document& oldDocument, Document& newDocument)
             }
         }
     }
+
+    if (langAttrKnownToMatchDocumentElement())
+        oldDocument.removeElementWithLangAttrMatchingDocumentElement(*this);
```

Comment:
FWIW the WeakHashSet lookup at this point is cheap because we've cleared it and the `find()` call will return early without even hashing the input value. But I can do this.

Reply:
I don't follow. We're just adopting this element to a new document. That document can have any number of existing elements with lang matching the document element, right?

Reply:
Sorry, I overlooked the word "new" in your comment, and you are right that we should be clearing the flag here.